frontend/web/components/pages/GitHubSetupPage.tsx

```diff
@@ -0,0 +1,286 @@
+              disabled
+              type='text'
+              title={'Repository Owner'}
+              inputClassName='input--wide'
+              placeholder='repositoryOwner'
+            />
+            <div style={{ width: '250px' }}>
+              <Select
+                size='select-md'
+                placeholder={'Select your repository'}
+                onChange={(v: repoType) => setRepositoryName(v.label)}
+                options={repositories?.repositories?.map((r: repoType) => {
+                  return { label: r.name, value: r.name }
+                })}
+              />
+            </div>
+            <Button
+              theme='primary'
+              disabled={!project || !installationId || !repositoryName}
+              onClick={() => {
+                {
+                  const newProjects = [...projects]
+                  newProjects.push(project)
+                  setProjects(newProjects)
+                }
+              }}
+            >
+              Add Project
+            </Button>
+          </Row>
+          <PanelSearch
+            className='no-pad mb-4'
+            title='Projects'
+            items={projects}
+            header={
+              <Row className='table-header'>
+                <Flex className='table-column px-3'>Project</Flex>
+                <div
+                  className='table-column text-center'
+                  style={{ width: '80px' }}
+                >
+                  Remove
+                </div>
+              </Row>
+            }
+            renderRow={(v: projectType) => (
+              <Row className='list-item' key={v.value}>
+                <Flex className='table-column px-3'>
+                  <div className='font-weight-medium mb-1'>{v.label}</div>
+                </Flex>
+                <div
+                  className='table-column  text-center'
+                  style={{ width: '80px' }}
+                >
+                  <Button
+                    onClick={() => {
+                      setProjects(
+                        projects.filter(
+                          (p: projectType) => p.value !== v.value,
+                        ),
+                      )
+                    }}
+                    className='btn btn-with-icon'
+                  >
+                    <Icon name='trash-2' width={20} fill='#656D7B' />
+                  </Button>
+                </div>
+              </Row>
+            )}
+          />
+          <Button
+            theme='primary'
+            disabled={!projects.length || !installationId || !repositoryName}
+            onClick={() => {
+              createGithubIntegration({
+                body: {
+                  installation_id: installationId,
+                },
+                organisation_id: organisation,
+              })
+            }}
+          >
+            Save Configuration
+          </Button>
+        </>
+      ) : (
+        <h1>Installation Completed</h1>
```

Comment:
Is this necessary ? When does this show, immediately as a window is closing ?

Reply:
Yes, it shows before the child window closes.